libs/s25main/EconomyModeHandler.cpp

```diff
@@ -22,181 +22,178 @@
 #include "GamePlayer.h"
 #include "GlobalGameSettings.h"
 #include "SerializedGameData.h"
+#include "helpers/containerUtils.h"
 #include "random/Random.h"
 #include "world/GameWorld.h"
 #include "world/GameWorldGame.h"
 #include "gameTypes/JobTypes.h"
 #include "gameData/JobConsts.h"
 #include <boost/optional.hpp>
 
-auto getPlayerMask = [](unsigned playerId) { return 1u << playerId; };
-
-GoodType types_less[]{
-  /*  1 */ GD_TONGS,      // Zange
-  /*  2 */ GD_HAMMER,     // Hammer
-  /*  3 */ GD_AXE,        // Axt
-  /*  4 */ GD_SAW,        // Saege
-  /*  5 */ GD_PICKAXE,    // Spitzhacke
-  /*  6 */ GD_SHOVEL,     // Schaufel
-  /*  7 */ GD_CRUCIBLE,   // Schmelztiegel
-  /*  8 */ GD_RODANDLINE, // Angel
-  /*  9 */ GD_SCYTHE,     // Sense
-  /* 12 */ GD_CLEAVER,    // Beil
-  /* 13 */ GD_ROLLINGPIN, // Nudelholz
-  /* 14 */ GD_BOW,        // Bogen
+std::array<GoodType, 12> specialGoodPool = {
+  /*  1 */ GD_TONGS,
```

Comment:
Could you either remove or update the index comments too? 1-9, then 12-14 doesn't make sense to me. I guess removing them is just fine.
PS: I'd really liked if C++ had a way to auto-deduce the size of a std::array but I guess that is C++17 with deduction guides only. Maybe a `makeArray` constexpr function would be good? It could be templated to take an array and deduce type and size and return a std::array. This way mistakes like having a size of 12 but 11 entries can be avoided.

Reply:
sure.

To the PS: I read that C++20 can auto-deduce the size from an initializer list (with std::to_array). For now I changed it to a const vector, because vector can auto-deduce the size. I know it isn't necessarily done at compile time then, but here I think it doesn't really matter either way.

Reply:
Actually `to_array` does more than that, the deduction guide rather acts like the now-removed `make_array`. I actually like that one, it seems to be pretty well and usable: https://godbolt.org/z/5x5xzE

Reply:
It's the least I can do. After all, I should have read the coding conventions for this project before making a pull request (I just found them). :D